### ISSUE TITLE
Error main.py

### ISSUE THREAD

**Issue by davidmilema** [RogerioBlanco/PixelCanvasBot#97]
![pixelcanvas_bot_error](https://user-images.githubusercontent.com/51240106/58716187-bf700c00-83c8-11e9-9a73-e76a5a017053.png)

No idea about this error.

**Comment by ZeedyDF** [comment]
You have to install the packages before trying to run the bot

Take a look here
https://github.com/RogerioBlanco/PixelCanvasBot#setup-bot

**Comment by davidmilema** [comment]
![pillowdavid](https://user-images.githubusercontent.com/51240106/58751652-63c58180-84a1-11e9-893b-4484c4c20092.png)

**Comment by davidmilema** [comment]
> You have to install the packages before trying to run the bot
> 
> Take a look here
> https://github.com/RogerioBlanco/PixelCanvasBot#setup-bot

Thank u, I installed the package, but now say this

![jsondavid](https://user-images.githubusercontent.com/51240106/58751740-0f230600-84a3-11e9-9755-7ded00610c61.png)

**Comment by Pr0phet-of-Fear** [comment]
This project is all but abandoned. You would be using [this version](https://github.com/Maphiu5/TraanvasBot) instead.

**Comment by davidmilema** [comment]
> This project is all but abandoned. You would be better off using [this version](https://github.com/Maphiu5/TraanvasBot) instead.

Wrong link

**Comment by Pr0phet-of-Fear** [comment]
Well it was the right link looks like it was taken down.

**Comment by RogerioBlanco** [comment]
I just want to clarify one thing: I don't fixed, yet, the bot because I don't have time for now, but anyone that wants fix I won't oppose

**Comment by TheRedMelon** [comment]
I cloned the fixed version while it was still up.
https://github.com/TheRedMelon/KekvasBot

**Comment by Pr0phet-of-Fear** [comment]
Is that Traanvas just renamed?

**Comment by TheRedMelon** [comment]
Yeah, not the most recent version before they made it private but it still works.

**Comment by Pr0phet-of-Fear** [comment]
Do you know why they took it down/made it private?

**Comment by TheRedMelon** [comment]
The commies and trannys in a war with Kekistan, they likely don't want us or anyone who would do them harm using it.  But I'd already cloned it and made some big improvements in a private repo.  And no, I'm not giving out those modifications I'm afraid.  I have no personal reason why I wouldn't want others to use this version though so I'm keeping it up.

**Comment by 4n0nct** [comment]
Yours has stop working too since yesterday:

`File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner`
`    self.run()`
`  File "/usr/lib/python3.7/threading.py", line 865, in run`
`    self._target(*self._args, **self._kwargs)`
`  File "/home/x/KekvasBot/src/bot.py", line 177, in update_canvas`
`    EnumColor.index(raw[index] >> 4))`
`IndexError: bytearray index out of range`

`Retried get_ws for time 1/4. JSONDecodeError: Expecting value: line 1 column 1 (char 0)`

Cloudflare changes again ?

**Comment by TheRedMelon** [comment]
No, complete overhaul basically.  I don't know enough to fix it though, I've never really done anything with networking before.  The requests go through but are unsuccessful.  I'll keep looking into it but you'll probably have to wait until someone else updates or releases one.

**Comment by 4n0nct** [comment]
I'm not good either for what's about networking, I'll wait and see, maybe traanvas will reappear with the magic trick once again like months ago.
Good continuation.

**Comment by Pr0phet-of-Fear** [comment]
I've been looking at the code, and the requests the site sends. I've tried some things, but haven't got anything to work. I don't really know enough about any of this to make a proper fix. If I do by some miracle make a fix, I will make pull requests.

What I know so far is that the api was moved from https://pixelcanvas.io/api to https://europe-west1-pixelcanvasv2.cloudfunctions.net/

**Comment by TheRedMelon** [comment]
'a' parameter has been replaced with 'wasabi'
the paths to the apis have been changed
pixelio.myself() isnt working
can't properly get the url for the websocket

**Comment by 4n0nct** [comment]
So, in the expectation of a solution, do one of you know any other fonctionnal bot ?
I tried the new dadido, but can't make it work on my penguin toaster.

**Comment by TheRedMelon** [comment]
Link to the bot thats been used in the recent EMF raids was posted by HF or whoever has that "PixelerM" alt in the discord. https://gitlab.com/Theemacsshibe/cl-pixelcanvas/tree/master/

**Comment by 4n0nct** [comment]
Thank you very much

**Comment by TheRedMelon** [comment]
On further inspection it looks like you need to login... I'm looking around and might see if I can figure out how to fix the old bot using this ones source code

**Comment by TheRedMelon** [comment]
Never mind, found the login: "GoogleMurrayBookchin"

**Comment by 4n0nct** [comment]
I feel retarded with portacle lel. It is mandatory to use it ?

**Comment by TheRedMelon** [comment]
uhh good luck trying to figure it out without, it's really easy just follow guide.

**Comment by 4n0nct** [comment]
I would like but...
![2019-06-19_00-01](https://user-images.githubusercontent.com/50212348/59723032-91425700-9225-11e9-9fcc-9b7adcf69c34.png)

**Comment by kubaello** [comment]
I've updated the URLs and the bot seems to be working again. #99

**Comment by 4n0nct** [comment]
That notification mail I have just received was a really good surprise.
Indeed, it works, thanks for your job

**Comment by Pr0phet-of-Fear** [comment]
[A surprise, to be sure, but a welcome one.](https://youtu.be/ca-e5MrVbVU)

**Comment by 4n0nct** [comment]
hum, there is still a disturbing thing.
It doesn't detect alreay painted pixel, it will put white on white for exemple, so for my actual drawing it has not evolved since 11 days cause it paints the pixels which are already here.
The status command confirm:
>already painted: 0px
>not painted: [the total number of px]

**Comment by Pr0phet-of-Fear** [comment]
Okay, so this probably deserves a new issue; but anyway, have you tried deleting the bot and redownloading it? I've been using the bot (well, yes, but actually no) since the update and encountered no such issue.

**Comment by 4n0nct** [comment]
That's what I did few minutes ago.
But...
![2019-08-13_16-58](https://user-images.githubusercontent.com/50212348/62952436-f06ab500-bdeb-11e9-85c5-b1075c2862a9.png)
Do I open a new issue ?
